pkg/registry/core/service/allocator/bitmap.go

```diff
@@ -84,6 +84,27 @@ func NewContiguousAllocationMap(max int, rangeSpec string) *AllocationBitmap {
 	return &a
 }
 
+// NewAllocationMapReserved creates an allocation bitmap using the random scan strategy
+// with some reserved offsets, so they can't be allocated randomly.
+func NewAllocationMapReserved(max int, rangeSpec string) *AllocationBitmap {
```

Comment:
I suggest placing this in a different file in the same package.  We have used `patch_bitmap.go` in the past.  This way the file will never conflict.